sdk/program/src/syscalls.rs

```diff
@@ -0,0 +1,123 @@
+use crate::{
+    instruction::{AccountMeta, ProcessedSiblingInstruction},
+    pubkey::Pubkey,
+};
+
+#[cfg(target_feature = "static-syscalls")]
+macro_rules! define_syscall {
+    (fn $name:ident($($arg:ident: $typ:ty),*) -> $ret:ty) => {
+		#[inline]
+        pub unsafe fn $name($($arg: $typ),*) -> $ret {
+			// this enum is used to force the hash to be computed in a const context
+			#[repr(usize)]
+			enum Syscall {
+				Code = sys_hash(stringify!($name)),
```

Comment:
This is cool, we sure this hash occurs at compile time?

Reply:
100%! An enum value must be known at compile time, and the resulting bytecode does contain CALL_IMM <hash>